### DISCUSSION TITLE
Bug with sorting

### DISCUSSION THREAD

**Discussion by ElGittoJunior** [livewire/livewire#7038] [7. Bugs (archived)]
### Livewire version

3.0.6 and 3.0.7

### Which PHP version are you using?

PHP 8.2

### Repository URL

https://github.com/ElGittoJunior/chirper-livewire-tutorial.git

### Steps To Reproduce

- I followed the official bootcamp.laravel.com tutorial for Livewire
- When I added editing functionality (https://bootcamp.laravel.com/livewire/editing-chirps), after clicking edit on a chirp in the dropdown menu the textbox appears, but the whole chirps list is sorted in reverse order. 
- If I save the chirp correct order is back, but when I press cancel its stays in the reverse order.

I noticed that its is working if i fix Livewire to 3.0.5, it also worked with 3.0.0 so I assume its working for versions 3.0.0-3.0.5

"require": {
        "php": "^8.1",
        "guzzlehttp/guzzle": "^7.2",
        "laravel/framework": "^10.10",
        "laravel/sanctum": "^3.2",
        "laravel/tinker": "^2.8",
        "livewire/livewire": "3.0.5",
        "livewire/volt": "^1.0"
    },

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by kresnasatya** [comment]
Hi, I'm also facing the same issue. I'm using Livewire version 3.0.10 and Livewire Volt version 1.4.0. I think the problem is with eloquent query. In Livewire Bootcamp to get chirps, we use `Chirp::with('user')->latest()->get()`, when we edit one of chirp, it will re-order the chirp although we already set `wire:key` property. I try to use `Chirp::with('user')->get()` without `latest()`, the issues gone. I'm curious why this happened and how to solve it?

**Reply by kresnasatya** [reply]
Finally, I figure out why this happened. I think because when we edit, cancel edit, or update the chirp, we don't call the `$this->getChirps()` method. This makes Livewire reset the collection order of chirps.

To solve this problem, we need to attach `$this->getChirps()` method when we edit, cancel edit, or update the chirp. It will even better, if we can use computed property. By using computed property, the data only load once and cache by Livewire. Then, we can refresh the data when chirp created, updated, or deleted by using `unset` keyword from PHP.

I already make the repo that solve the issue. Feel free to take a look.

Livewire Functional: https://github.com/kresnasatya/chirper/tree/livewire-functional

Livewire Class: https://github.com/kresnasatya/chirper/tree/livewire
